Plutarch/Api/V1/Value.hs

```diff
@@ -48,3 +57,102 @@ deriving via
   (DerivePConstantViaNewtype Plutus.Value PValue (PMap PCurrencySymbol (PMap PTokenName PInteger)))
   instance
     PConstantDecl Plutus.Value
+
+-- | Works correctly only on 'normalize'd and sorted values!
+instance PEq PValue where
+  a #== b = pto a #== pto b
+
+instance Semigroup (Term s PValue) where
+  a <> b = unionWith # plam (+) # a # b
+
+instance Monoid (Term s PValue) where
+  mempty = pcon (PValue AssocMap.empty)
+
+-- | Construct a singleton 'PValue' containing only the given quantity of the given currency.
+singleton :: Term (s :: S) (PCurrencySymbol :--> PTokenName :--> PInteger :--> PValue)
```

Comment:
I'd prefer a version that takes in `PAsData` values instead, either as a separate function or a modified version of this one.

The same goes for any other function that could skip a `pdata`/`pfromData` within if provided the more focused type.

Reply:
There already is `singletonData`.

Reply:
Not for `Value` though, but how often do you have `Term s (PAsData PInteger)` lying around? It seems unlikely.

Reply:
Very often if it's in your datum or redeemer?

Reply:
I've added the function.